### PR TITLE
Add lake package + example to nlmod

### PR DESCRIPTION
**Pull request by OnnoEbbens**

### CODE REVIEW

**Review by rubencalje — CHANGES_REQUESTED** [review]
See comments

File: nlmod/gwf/lake.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+    """add a lake from a geodataframe
+
+    Parameters
+    ----------
+    gwf : flopy.mf6.modflow.mfgwf.ModflowGwf
+        groundwater flow model.
+    gdf : gpd.GeoDataframe
+        geodataframe with the cellids as the index and the columns:
+            lakeno : with the number of the lake
+            elev : with the bottom of the lake
```
[the remark this answers is not in the document]
Comment (OnnoEbbens) [reply nlmod/gwf/lake.py:53]: True

File: nlmod/gwf/lake.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+    perioddata = {}
+    for iper in range(ds.dims["time"]):
+        perioddata[iper] = []
+
+    lake_settings = []
+    for setting in LAKE_KWDS:
+        if setting in gdf.columns:
+            lake_settings.append(setting)
+
+    lake_settings = [setting for setting in LAKE_KWDS if setting in gdf.columns]
```
Comment (dbrakenhoff) [inline nlmod/gwf/lake.py:113]: code doing the same thing twice?

Comment (OnnoEbbens) [reply nlmod/gwf/lake.py]: check

**Pull request merged**